prime-router/docs/features/0004-managing-translation-schemas.md

```diff
@@ -0,0 +1,217 @@
+    - `/v1/senders|receivers/{sender|receiverName}/schemas/{transformType}/{*schemaName}`
+
+#### Common schemas
+- Create a common schema
+    - POST
+    - Auth: Prime admins
+    - `/v1/schemas/{transformType}`
+    - Body
+        - `schemaName` string - optional
+            - If the field is set also update the sender or receiver setting
+        - `schema` string
+- Update a common schema
+    - PUT
+    - Auth: Prime admins
+    - `/v1/schemas/{transformType}/{*schemaName}`
+    - Body
+        - `schema` string
+- Rollback a common schema
+    - Rolls back to the previous version of that schema name, if the rolling back the first version, the file is deleted and the schemaName field is removed from the
+    - DELETE
+    - Auth: Prime admins
+    - `/v1/senders|receivers/{sender|receiverName}/{transformType}/{*schemaName}`
+- Invalidate cache
+    - Invalidates all the cached values for the particular transform type; would be potentially need to get invoked if someone edited the schema in the azure store
+      - Future iterations could make this invalidation more fine-grained
+    - GET
+    - Auth: Prime admins
+    - `/v1/schemas/{transformType}`
+
+### Validation
+
+As part of either creating or updating any schema, it will need to be validated before it can get persisted. Every schema that is uploaded will check that the YAML can be parsed and that resolving it does not throw a `SchemaException` that are currently thrown when parsing a schema by trying to resolve it.
+Additionally, the APIs for changing schemas for a sender or receiver can accept a sample input and output message that will be used to verify that the schema works as expected.
+
+Future iterations could improve upon this:
+
+- Provide the specific line that is invalid
+
+## Operations
+
+### Local Development
+
+Once the migration to store the schemas as data rather that source code is done, there will still need a solution in place that enables local developers to submit reports through the universal pipeline with sender/receiver transforms applied.  The easiest solution is to update the `reloadSettings` gradle task (or add a new `reloadSchemas`) that invokes the create schema APIs for some of the ignore senders and receivers.
+
+### Testing
+
+This is area that is impacted the most by shifting from treating schemas as source code to data.  The current approach when working on a schema change is to make the change in the source code and then update the output file to reflect what the schema change should now do and these integration tests are run as part of CI builds to verify that they continue to work as expected.  This approach will no longer work effectively once the schemas are editable via the application; an example issue that could occur: an engineer is iterating on a schema change for an STLT, while going through these iterations, the schema test for that STLT will break and pull requests will be stuck.
+
+Some potential solutions (several could be used):
```

Comment:
I'll add that as another possibility too, though my main concern is the amount of overhead.  I'm also not convinced that there would be a justification to run this during CI since we might get a lot of false positives on failures